### ISSUE TITLE
FEATURE REQUEST: supporting the genuine tibber price sensor provided by the tibber integration

### ISSUE THREAD

**Issue by typxxi** [kotope/aio_energy_management#25]
I have been facing issues with nordpool and entso-e data or price quality and therefore I am asking for a simple way to use AIO with tibber HA sensors too like the nordpool / entso-e editions.

I have been using nordpool for months with 0 deviation to the tibber price based on their app and their tibber integraiton sensor which is now working again (the mathplot error from spring 2024 has been resolved at least after I had reinstalled the integration this week again). I have used the same old nordpool modifyer or calculation for vat and fees which had lead to a deviation since monday of up to 7 Cents / kWh if you compare nordpool price to tibber price (39,8 cent versus 47,x cent is inacceptable).

This is the template I had been using all the time : 20,72 cent were the fees on top of the 19% vat on the electrity price
`{{ (0.2072 + (current_price * 0.19)) | float }}`

Today I went on and did the entso-e integration and again a gap even with the proper modifier using the "net fees" = 0.2072€ / 1,19 = 0.174117647059  
```
{% set s = {
  "distribution": 0.0,
  "electricity_tax": 0.0,
  "variable_network_fees": 0.0,
  "concession_fees": 0.0,
  "kwk": 0.0,
  "nev": 0.0,
  "offshore": 0.0,
  "ablav": 0.0,
  "guarantees_of_origin": 0.0,
  "misc": 0.174117647059
}
%}
{{(current_price + s.distribution + s.electricity_tax + s.variable_network_fees + s.concession_fees + s.kwk + s.nev + s.offshore + s.ablav + s.guarantees_of_origin + s.misc) | float}}
```
And even now the gap is still 1,5 Cent at a lowest price level of today and I can never be sure that this value might increase or drift away. 

I did dive deep here in a discussion where I describe(d) my "journey" to find a proper solution incl. a lot of screenshots to make clear that those tibber prices are the right ones (provided by tibber app and also by the HA tibber integration sensor).

https://github.com/kotope/aio_energy_management/discussions/24

Maybe my modifiyers are wrong and I missed a break like that by nordpool on monday when they claimed that they had only changed the api and we should only have to change from location or region "DE-LU" aka german-luxemburg to "GER". But that was not true and i have spend 3 days with this whole topic to establish a "platform" which I could use for AIO. But both are not working "properly" enough even though at least entsoe is way less off or I have to investigate the whole nordpool mess regarding a change in the modifier template structure. 

Tibber is here the next big thing, people get solar power, many build their own battery and then tibber becomes the obvious next step. 

This is how the tibber price sensor looks like
![image](https://github.com/user-attachments/assets/c6163694-8a7a-4315-8270-14e7088e9eb8)

and here is the data structure (where tomorrow is missing cause those prices will be released in 1 or 2 hours)
```
today: 
- total: 0.2118
- total: 0.2059
- total: 0.204
- total: 0.1982
- total: 0.2034
- total: 0.2269
- total: 0.2754
- total: 0.3121
- total: 0.3016
- total: 0.2793
- total: 0.2655
- total: 0.2534
- total: 0.2469
- total: 0.248
- total: 0.2629
- total: 0.2671
- total: 0.3018
- total: 0.3386
- total: 0.4131
- total: 0.3964
- total: 0.3313
- total: 0.3079
- total: 0.2975
- total: 0.2893

tomorrow: 
unit_of_measurement: EUR/kWh
friendly_name: Tibber Prices
``` 

I have no clue how much work that might be to support tibber but it is the biggest player in the german (speaking) market .

**Comment by typxxi** [comment]
No longer need cause I have finally fixed the whole thing.

I had been using Nordpool provided by your blog and there you had explained back in 2022 or so how to set it up via sensors.yaml file which I did.

the new version of 0.0.15 now introduced new regions and by default an enabled VAT option in the dialog box.

That went wrong cause that introduced or brought me into a bug they had caused cause for GER they had used the 0.23 VAT figure instead of 0.19 for 19% . Before monday it had worked out fine for me cause I had been using the proper codes for additional costs ... but with the upgrade to 0.0.15 the big mess had begun and I was never aware of the question that they were not sure about german VAT and had instead simply used 23% which no german users would have had expect. 

My deviation is now zero cent cause there is none as before ...

and therefore I can start the AIO 0.2.2 adventure soon.

thanks

**Comment by typxxi** [comment]
for german users [here ](https://github.com/custom-components/nordpool/issues/422) is the fix of the nordpool integration 0.0.15 with wrong VAT figure of 23% instead of 19%.

**Comment by kotope** [comment]
Good to hear you found the root cause!

Anyhow, I briefly checked out Tibber yesterday as I though it could be anyhow a good feature to add.
However it seemed that the Tibber is not available in Finland so I would not be able to implement it myself at least.
